transactions/transactor_test.go

```diff
@@ -385,41 +401,152 @@ func (s *TxQueueTestSuite) TestLocalNonce() {
 	s.Equal(uint64(nonce)+1, resultNonce.(uint64))
 }
 
-func (s *TxQueueTestSuite) TestContractCreation() {
-	key, _ := crypto.GenerateKey()
-	testaddr := crypto.PubkeyToAddress(key.PublicKey)
-	genesis := core.GenesisAlloc{
-		testaddr: {Balance: big.NewInt(100000000000)},
-	}
-	backend := backends.NewSimulatedBackend(genesis)
-	selectedAccount := &account.SelectedExtKey{
-		Address:    testaddr,
-		AccountKey: &keystore.Key{PrivateKey: key},
-	}
-	s.manager.sender = backend
-	s.manager.gasCalculator = backend
-	s.manager.pendingNonceProvider = backend
-	tx := SendTxArgs{
-		From:  testaddr,
-		Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+func (s *TxQueueTestSuite) TestTransactionsWithSimulatedBackend() {
+
+	testCases := []struct {
+		name             string
+		getTransactions  func(testaddr common.Address) []SendTxArgs
+		receiptValidator func(receipt *types.Receipt, finalResult sign.Result, testaddr common.Address)
+	}{
+		{
+			"EthTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Value: (*hexutil.Big)(big.NewInt(10)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, finalResult sign.Result, testaddr common.Address) {
+				successStatus := uint(1)
+				s.Equal(successStatus, receipt.Status)
+			},
+		},
+		{
+			"ContractCreation",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, finalResult sign.Result, testaddr common.Address) {
+				s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
+			},
+		},
+		{
+			"ContractStateTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				// method ID: web3.utils.sha3('setOwner(bytes32,address)').substring(0, 8)
+				setOwnerMethodID := "0x5b0fc9"
+				node := make([]byte, 32)
+				// Give the node a value so that we are reassigning ownership of something
+				// other than the 0 node, which already belongs to the current owner
+				node[0] = []byte("1")[0]
```

Comment:
you can also do [32]byte{1} , which will serve the purpose. or if you want to use slice
for some reason you still can just set any element without conversions